src/main/java/org/veupathdb/service/eda/ss/model/reducer/ValueExtractor.java

```diff
@@ -0,0 +1,91 @@
+package org.veupathdb.service.eda.ss.model.reducer;
+
+import org.veupathdb.service.eda.ss.model.variable.VariableValueIdPair;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Provides a stream of variable value records extracted from a stream of ID indexes and a collection of
```

Comment:
What is a "variable value record?"   seems it is a List<String>.   what are those strings?  comment would help

Reply:
There are issues with how this works now. I need to follow up on this when I hook this up end-to-end. I'll add a comment on what it is now and appropriate TODOs